### DISCUSSION TITLE
Why Skybox's geometry only uses three vertices?

### DISCUSSION THREAD

**Discussion by zy4944796** [google/filament#6970] [Q&A] [answered]
A cube requires 6 faces, 12 triangles, and thus 36 vertices. However, in the geometry used for skyboxes, the vertex array obtained using getFullScreenVertexBuffer only has three vertices.
![Screenshot from 2023-07-20 14-58-55](https://github.com/google/filament/assets/8578753/5f0794ef-21e1-4174-927c-3529ee3e0688)

![Screenshot from 2023-07-20 14-59-55](https://github.com/google/filament/assets/8578753/520830f5-5d08-44df-9c96-4c1e23d64bd3)
![Screenshot from 2023-07-20 15-03-07](https://github.com/google/filament/assets/8578753/a176e451-2d23-4ccc-abd5-f809f3c332ec)

**Comment by romainguy** [comment, marked as answer]
There's no need to build a cube. We use a single triangle that covers the entire screen and compute the view direction at each fragment in the shader (we do this in the vertex shader). That view direction is all that's needed to sample the cubemap.

**Reply by zy4944796** [reply]
Thank you for your explanation, I understand now.
